### DISCUSSION TITLE
По поводу тем.

### DISCUSSION THREAD

**Discussion by Survolog** [anilibria/anilibria-winmaclinux#102] [Идеи]
Красный цвет это прекрасно и всё такое.
Но (мне) хотелось бы спокойный синеватый пастельный оттенок интерфейса.
Чтобы по глазам не бил.

**Comment by trueromanus** [comment]
Здравствуйте, в версии 1.2.1 уже добавилась возможность создавать свои темы. Там пока не все конечно, но цвета многих элементов менять уже можно. В следующих версиях она будет расширятся. Вы можете уже попробовать поменять цвет.

**Reply by Survolog** [reply]
Да-да. Я просто по-привычке не догадался хотя бы открыть справку. Она помогла найти, как менять цвет.
Хотелось бы на меню справа сверху, в котором выбирается "Редактор темы", иметь значёк гаечного ключа или шестерёнки. То, что квадратный значок из четырёх синих квадратиков открывает редактирование, я не смог сам догадаться.

В меню выбора цвета не хватает отображения цвета по умолчанию. Налочка стоит всегда на чёрном цвете. Возможно, вместо "Не перекрыт" лучше бы сразу рисовало используемый цвет или писало его код. Всё-таки, белый плюс на зелёном фоне уже даёт информацию, что цвет не перекрыт.

Долго вставлять один и тот же цвет. Возможно, стоило добавить выбор нескольких позиций.
Или пстоило оделить список на тематические группы.
В идеале не хватает генерации связанных цветов.
Например, тень могла бы сразу отличаться на столько-то процентов от выбранного общего цвета, фон нажатого элемента - на столько-то, фон выделенного элемента - на столько-то.
Если подобрать коэффициенты, то не думаю, что кто-то стал бы менять остальные связанные цвета.

При нажатии на "Создать пустую тему" что-то не меняется ничего. Только "Копировать из темы" у меня сработало.
Получается, что хотя бы одну тему (зелёную) необходимо скачать.
Может это баг лично моей сборки пакета.

Удаляемые темы исчезают не сразу.
Возможны темы с одинаковыми названиями.
Сохранить тему в файл не смог. В терминале не увидел каких-либо сообщений по этому поводу.
Может это связано.
Опять таки может это баг лично моей сборки пакета.

Оказалось не очень просто найти конфиг с темой. Он не в ~/.config/anilibria
Но это не похоже на проблему.
Нашёл сравнением ~/ до и после редактирования темы.
`~/.local/share/EmptyFlow/AnilibriaDesktopClient/applicationtheme.cache`
Не претендую ни на что. Сделал пока себе так:
```
        {
            "basedOnTheme": "Светлая",
            "drawerGradiendStep0": "#78b0db",
            "drawerGradiendStep1": "#78b0db",
            "drawerGradiendStep2": "#78b0db",
            "drawerGradiendStep3": "#78b0db",
            "drawerGradiendStep4": "#78b0db",
            "name": "Blue2",
            "pageBackground": "#78b0db",
            "pageVerticalPanel": "#78b0db"
        },

```
Конечно, хотелось бы покрасивше. Например, не хватает замены какого-то из цветов на более тёмный (или более светлый), чтобы вернулась переливчатость. Но на мой вкус уже стало спокойнее.

Спасибо за очень удобную программу. Если и есть что-то более удобное не просто для просмотра мультиков, но для просмотра тв или кино через интернет, я подобного не встречал.

**Reply by trueromanus** [reply]
>В меню выбора цвета не хватает отображения цвета по умолчанию. Налочка стоит всегда на чёрном цвете. Возможно, вместо "Не перекрыт" лучше бы сразу рисовало используемый цвет или писало его код. Всё-таки, белый плюс на зелёном фоне уже даёт информацию, что цвет не перекрыт.  
 
В следующей версии будет предпросмотр цвета и указание что цвета нет
![image](https://user-images.githubusercontent.com/9040186/203844155-f6a88d8b-aef4-4cf6-8ca5-004c63a74e5f.png)

> Долго вставлять один и тот же цвет. Возможно, стоило добавить выбор нескольких позиций.
> Или пстоило оделить список на тематические группы.
> В идеале не хватает генерации связанных цветов.
> Например, тень могла бы сразу отличаться на столько-то процентов от выбранного общего цвета, фон нажатого элемента - на > столько-то, фон выделенного элемента - на столько-то.
> Если подобрать коэффициенты, то не думаю, что кто-то стал бы менять остальные связанные цвета."    

Интересная идея подумаю над этим.

> При нажатии на "Создать пустую тему" что-то не меняется ничего. Только "Копировать из темы" у меня сработало.
Получается, что хотя бы одну тему (зелёную) необходимо скачать.
Может это баг лично моей сборки пакета.

Создать пустую тему должна только очищать поля, она и этого не делает?
> Удаляемые темы исчезают не сразу.
Возможны темы с одинаковыми названиями.
Сохранить тему в файл не смог. В терминале не увидел каких-либо сообщений по этому поводу.
Может это связано.
Опять таки может это баг лично моей сборки пакета.

Нет баги в коде, уже поправлены и в следующей версии такого беспредела не будет. Редактирование тем это новый функционал который заехал буквально в последнюю версию поэтому есть шероховатости :( Извиняюсь за неудобства.

> Оказалось не очень просто найти конфиг с темой. Он не в ~/.config/anilibria
Но это не похоже на проблему.
Нашёл сравнением ~/ до и после редактирования темы.

Вроде и не предполагалось что ее надо искать, сохранение в файл должно было решить эту проблему :(

> Конечно, хотелось бы покрасивше. Например, не хватает замены какого-то из цветов на более тёмный (или более светлый), чтобы вернулась переливчатость. Но на мой вкус уже стало спокойнее.

Понял, идея интересная тоже добавлю в список. Тут главное чтобы вкус был хотя бы частично удовлетворен 😉 

Спасибо Вам за расширенный ответ и интересные предложения 👍
